lib/pos_epoch_complete_hook.go

```diff
@@ -119,50 +189,16 @@ func (bav *UtxoView) RunEpochCompleteHook(blockHeight uint64) error {
 		return validatorSetPKIDs.Includes(*s.ValidatorPKID)
 	})
 
-	// Snapshot only the top n stake entries that are in the validator set.
+	// Snapshot only the top m stake entries that are in the validator set.
 	for _, stakeEntry := range topStakesInValidatorSet {
 		snapshotStakeEntry := SnapshotStakeEntry{
-			SnapshotAtEpochNumber: currentEpochEntry.EpochNumber,
+			SnapshotAtEpochNumber: epochNumber,
 			ValidatorPKID:         stakeEntry.ValidatorPKID,
 			StakerPKID:            stakeEntry.StakerPKID,
 			StakeAmountNanos:      stakeEntry.StakeAmountNanos,
 		}
 		bav._setSnapshotStakeToReward(&snapshotStakeEntry)
 	}
 
```

Comment:
so we are no longer pruning old snapshots. This is fine for now I think.

Reply:
We never pruned old snapshots. It's still TBD whether/when it's safe to prune them, so we're keeping them indefinitely for now